modules/core/src/utils/typed-array-manager.js

```diff
@@ -39,24 +48,28 @@ export class TypedArrayManager {
     this._release(typedArray);
   }
 
-  _allocate(Type, size, initialize) {
+  _allocate(Type, size, initialize, maxSize) {
     // Allocate at least one element to ensure a valid buffer
-    size = Math.max(Math.ceil(size * this.props.overAlloc), 1);
+    let sizeToAllocate = Math.max(Math.ceil(size * this.props.overAlloc), 1);
+    // Don't over allocate after certain specified number of elements
+    if (sizeToAllocate > maxSize) {
+      sizeToAllocate = Math.max(size, maxSize);
```

Comment:
```suggestion
      sizeToAllocate = maxSize;
```

Reply:
This is not what I intended to do originally with the overAllocCountCap.
It isn't maxSize for the allocation, it is a limit after which the over-allocation stops.
But we at the same time we always get enough memory allocated to fit **count** of elements.

Reply:
Then you need to change `maxCount` from `MAX_PICKING_COLOR_CACHE_SIZE` to `Math.max(numInstances, MAX_PICKING_COLOR_CACHE_SIZE)`.